synapse/lib/cell.py

```diff
@@ -2585,18 +2604,25 @@ async def _bootCellProv(self):
 
         return provconf
 
-    def _bootCellConf(self, conf=None):
-        # create a fresh config and re-resolve inputs in precedence order...
-        conf = s_config.Config.getConfFromCell(self, conf=conf)
+    async def _bootCellConf(self, conf=None):
+        new_conf = self.initCellConf()
+        new_conf.setdefault('_log_conf', await self._getSpawnLogConf())
+
+        new_conf.setConfFromOpts(self.conf.getOpts())
+        new_conf.setConfFromEnvs()
 
         path = s_common.genpath(self.dirn, 'cell.yaml')
+        s_common.yamlmod(conf, path)
```

Comment:
Wouldn't this still save env vars in `cell.yaml` which we were trying to avoid?

Reply:
This value is the provisioning config data, which wouldn't contain data from envars.
I'm reworking this to be more explicit about ordering / variable naming since things are currently duplicated and confusing to read.

Reply:
This value is the provisioning config data, which wouldn't contain data from envars.
I'm reworking this to be more explicit about ordering / variable naming since things are currently duplicated and confusing to read.